### PR TITLE
fix: HFP occupancy status is set EMPTY when occu is zero

### PR DESCRIPTION
**Pull request by thjarvin**

### CODE REVIEW

**Comment by thjarvin** [comment]
Tämä korjaus pyrkii korjaamaan graylogissa usein näkyvän virheen "Exception while handling message" ja siihen liittyvän poikkeuksen: "java.lang.NullPointerException: null"

**Review by LofhJann — APPROVED** [review]

File: src/main/java/fi/hsl/transitdata/vehicleposition/application/gtfsrt/GtfsRtOccupancyStatusHelper.java [hunk 1]
```diff
@@ -60,6 +60,8 @@ public Optional<GtfsRealtime.VehiclePosition.OccupancyStatus> getOccupancyStatus
             //Currently occu is only available for Suomenlinna ferries
             if (hfpPayload.getOccu() != 0) {
```
Comment (haphut) [inline src/main/java/fi/hsl/transitdata/vehicleposition/application/gtfsrt/GtfsRtOccupancyStatusHelper.java:61]: Are negative values possible? Should this be 
```suggestion
            if (hfpPayload.getOccu() > 0) {
```

File: src/main/java/fi/hsl/transitdata/vehicleposition/application/VehiclePositionHandler.java [hunk 2]
```diff
@@ -182,18 +182,14 @@ public void handleMessage(Message message) {
...
-                    detailMessage = "stopStatusProcessor.getStopStatus(data) called";
                     String uniqueVehicleId = getUniqueVehicleId(data.getTopic().getOperatorId(), data.getTopic().getVehicleNumber());
-                    detailMessage = "getUniqueVehicleId called";
                     PassengerCount.Payload passengerCount = passengerCountCache.getPassengerCount(uniqueVehicleId, data.getPayload().getRoute(), data.getPayload().getOday(), data.getPayload().getStart(), data.getPayload().getDir());
-                    detailMessage = "passengerCountCache.getPassengerCount called";
+                    
                     if (!isValidPassengerCountData(passengerCount)) {
                         if (passengerCount != null) {
-                            log.warn("Passenger count for vehicle {} was invalid (vehicle load: {}, vehicle load ratio: {})",
+                            log.debug("Passenger count for vehicle {} was invalid (vehicle load: {}, vehicle load ratio: {})",
```
Comment (haphut) [inline src/main/java/fi/hsl/transitdata/vehicleposition/application/VehiclePositionHandler.java:192]: I don't know what the process is for catching bugs in the onboard APC devices. Would turning this into debug effectively silence it? Do we have a feedback loop elsewhere to communicate with the APC device vendor?

Comment (haphut) [reply src/main/java/fi/hsl/transitdata/vehicleposition/application/VehiclePositionHandler.java:192]: Or maybe it's not a bug for vehicleLoad to be negative as there is measurement error anyway?

**Comment by haphut** [comment]
Otherwise, looks good to me. Well done!

File: src/main/java/fi/hsl/transitdata/vehicleposition/application/VehiclePositionHandler.java (hunk 2 above)
Comment (thjarvin) [reply src/main/java/fi/hsl/transitdata/vehicleposition/application/VehiclePositionHandler.java:192]: log.debug was used before I added logging for debugging purposes

File: src/main/java/fi/hsl/transitdata/vehicleposition/application/gtfsrt/GtfsRtOccupancyStatusHelper.java (hunk 1 above)
Comment (thjarvin) [reply src/main/java/fi/hsl/transitdata/vehicleposition/application/gtfsrt/GtfsRtOccupancyStatusHelper.java]: I made the change you suggested and added a unit test as well.

**Review by haphut — APPROVED** [review]

**Pull request merged**